yorkie/src/main/kotlin/dev/yorkie/document/crdt/CrdtArray.kt

```diff
@@ -1,3 +1,162 @@
+    // `purge` physically purge child element.
+    /**
+     * Physically purges the given [element].
+     */
+    override fun purge(element: CrdtElement) {
+        elements.purge(element)
+    }
+
+    // NOTE(7hong13): Original comment from JS SDK is as follows:
+    // `insertAfter` adds a new node with the value after the given node.
+    /**
+     * Adds a new node with [value] after the node created at [prevCreatedAt].
+     */
+    fun insertAfter(prevCreatedAt: TimeTicket, value: CrdtElement) {
+        elements.insertAfter(prevCreatedAt, value)
+    }
+
+    /**
+     * Moves the given [createdAt] element after the [prevCreatedAt] element.
+     */
+    fun moveAfter(prevCreatedAt: TimeTicket, createdAt: TimeTicket, executedAt: TimeTicket) {
+        elements.moveAfter(prevCreatedAt, createdAt, executedAt)
+    }
+
+    /**
+     * Returns the element of the given [createdAt].
+     */
+    fun get(createdAt: TimeTicket): CrdtElement? {
+        val node = elements.get(createdAt)
+        if (node?.isRemoved == true) return null
+        return node
+    }
+
+    /**
+     * Returns the element of the given [index].
+     */
+    fun getByIndex(index: Int): CrdtElement? {
+        val node = elements.getByIndex(index)
+        return node?.value
+    }
+
+    // NOTE(7hong13): Original comment from JS SDK is as follows:
+    // `getPrevCreatedAt` returns the creation time of the previous element of the given element.
+    /**
+     * Returns a creation time of the previous node.
+     */
+    fun getPrevCreatedAt(createdAt: TimeTicket): TimeTicket {
+        return elements.getPrevCreatedAt(createdAt)
+    }
+
+    // NOTE(7hong13): Original comment from JS SDK is as follows:
+    // `delete` deletes the element of the given index.
+    /**
+     * Deletes the node of the given creation time.
+     */
+    override fun delete(createdAt: TimeTicket, executedAt: TimeTicket): CrdtElement {
+        return elements.delete(createdAt, executedAt)
+    }
+
+    /**
+     * Deletes the element of given [index] and [editedAt].
+     */
+    fun deleteByIndex(index: Int, editedAt: TimeTicket): CrdtElement? {
+        return elements.deleteByIndex(index, editedAt)
+    }
+
+    /**
+     * Traverses the descendants of this array.
+     */
+    override fun getDescendants(callback: (CrdtElement, CrdtContainer) -> Boolean) {
+        for (node in elements) {
+            val element = node.value
+            if (callback(element, this)) return
+
+            if (element is CrdtContainer) {
+                element.getDescendants(callback)
+            }
+        }
+    }
+
+    /**
+     * Returns the JSON encoding of this array.
+     */
+    override fun toJson(): String {
+        val json = mutableListOf<String>()
+        return forEach { json.add(it.toJson()) }.run {
+            json.joinToString(",")
+        }
+    }
+
+    // TODO(7hong13): implement toJS method (json parser)
```

Comment:
Currently, we use the comment like `TODO($name)` when developer need to implement the function body. 
But, I think, we would use `TODO(String)` function would rather than `TODO($name)` comment for more formal way. Because it can throw the Exception if it wasn't implemented.